controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -73,6 +73,106 @@ static bool validate_hex_notation(const std::string &str, uint8_t expected_octet
     return true;
 }
 
+/**
+ * @brief Parse bss_info string into bss_info_conf_struct.
+ * 
+ * @param[in] bss_info_str String conataining bss info configuration
+ * @param[out] bss_info_conf Controller database struct filled with configuration.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+static std::string parse_bss_info(const std::string &bss_info_str,
+                                  db::bss_info_conf_t &bss_info_conf, std::string &err_string)
```

Comment:
Maybe it's better to pass al_mac as an `[out]` parameter and return bool. The doxygen claims that it return bool, BTW :-).

It may be a good idea if we would converge on the pattern to use for any function. Now, we have some functions that return a result which may have a sentinel value to indicate error (like in this case, the empty string), and other functions that return bool and take arguments for their outputs.

Reply:
I would prefer to avoid as much as I can from getting the result from a function argument.
I hate when I have to declare a return variable before calling to a function.
But I'll change the doxygen return.

Reply:
agree with Moran on this one, I also hate it when I have to declare an extra variable so if it can be avoided like here when we return an empty string, then I prefer that.

Reply:
We're deciding on a lot of this kind of coding rules in our reviews, so I've updated the CODINGSTYLE.md with the ones I remember: #242 

It would be nice if others would also update CODINGSTYLE.md like that.